### PR TITLE
Several improvements

### PR DESCRIPTION
**Pull request by Laurensvanrun**
-Win64 and WOW64 support
-JCL updated
-Better coverage on generic classes
-EPrivilegeExceptions and EExternalExceptions fix (multithreading problem)

### CODE REVIEW

File: Source/CodeCoverage.dproj [hunk 1]
```diff
@@ -45,12 +40,6 @@
         <Cfg_2>true</Cfg_2>
         <Base>true</Base>
     </PropertyGroup>
-    <PropertyGroup Condition="('$(Platform)'=='Win64' and '$(Cfg_2)'=='true') or '$(Cfg_2_Win64)'!=''">
-        <Cfg_2_Win64>true</Cfg_2_Win64>
-        <CfgParent>Cfg_2</CfgParent>
-        <Cfg_2>true</Cfg_2>
-        <Base>true</Base>
-    </PropertyGroup>
```
Comment (ekot1) [inline Source/CodeCoverage.dproj:53]: Looks like 64 bit configuration got deleted by accident

File: Source/Debugger.pas [hunk 2]
```diff
@@ -991,8 +1003,15 @@ procedure TDebugger.LogStackFrame(const ADebugEvent: DEBUG_EVENT);
...
   Module: IDebugModule;
   MapScanner: TJCLMapScanner;
+  MachineType: Cardinal;
 begin
   ContextRecord.ContextFlags := CONTEXT_ALL;
+  case FProcessTarget of
+    taWin32:
+      MachineType := IMAGE_FILE_MACHINE_I386;
+    taWin64:
+      MachineType := IMAGE_FILE_MACHINE_AMD64;
```
Comment (ekot1) [inline Source/Debugger.pas:1028]: This should have an else branch or compiler shows warning. Maybe 
  else
  begin
      FLogManager.Log('Unkown platform');
      Exit;
  end;

**Review by ekot1 — CHANGES_REQUESTED** [review]

File: Source/CodeCoverage.dproj (hunk 1 above)
Comment (Laurensvanrun) [reply Source/CodeCoverage.dproj:53]: Whoops :-)

File: Source/Debugger.pas (hunk 2 above)
Comment (Laurensvanrun) [reply Source/Debugger.pas:1028]: Yeah, I was lazy, sorry my bad, solved.

**Pull request merged**